pkg/cmd/cli/restic/init_repository_test.go

```diff
@@ -0,0 +1,65 @@
+package restic
+
+import (
+	"testing"
+
+	"github.com/spf13/pflag"
+	"github.com/stretchr/testify/assert"
+
+	"k8s.io/client-go/kubernetes"
+
+	"github.com/heptio/ark/pkg/client"
+	clientset "github.com/heptio/ark/pkg/generated/clientset/versioned"
+)
+
+type fakeFactory struct{}
+
+var _ client.Factory = &fakeFactory{}
+
+func (f *fakeFactory) BindFlags(flags *pflag.FlagSet) {
+	panic("not implemented")
+}
+
+func (f *fakeFactory) Client() (clientset.Interface, error) {
+	panic("not implemented")
+}
+
+func (f *fakeFactory) KubeClient() (kubernetes.Interface, error) {
+	panic("not implemented")
+}
+
+func (f *fakeFactory) Namespace() string {
+	return ""
+}
+
+func TestComplete(t *testing.T) {
+	// no key options provided should error
+	o := &InitRepositoryOptions{}
+	assert.NotNil(t, o.Complete(&fakeFactory{}))
+
+	// both KeyFile and KeyData provided should error
+	o = &InitRepositoryOptions{
+		KeyFile: "/foo",
+		KeyData: "bar",
+	}
+	assert.NotNil(t, o.Complete(&fakeFactory{}))
+
+	// TODO test that if KeyFile is provided, the data is used
```

Comment:
+1

Reply:
OK with me just doing this refactor as part of this PR?

Reply:
Yes